operator/src/main/resources/scripts/utils_base.sh

```diff
@@ -357,6 +357,37 @@ checkSourceWDTInstallDirExistsAndNotEmpty() {
   fi
 }
 
+#
```

Comment:
The legacy "initAuxiliaryImage" function was not reverted back to the original? It looks like it has been repurposed and is now used for the new approach. I think this risks pods that started before the operator was converted -- they will have their scripts changed out from underneath them, leading to unpredictable behavior. The old functions should all be left in place and unchanged - plus hopefully leveraged for the 'converted' domains. New functions should be added for handling the new approach -- instead of re-using the function names of the old approach.

Reply:
I'm not sure if I understand this correctly. I have tested the scripts with both the 3.x auxiliary images and new 4.0 auxiliary images and it works fine in both cases. The 3.x auxiliary image pods will always be converted to "raw" init containers using special names (starting with compatibility-mode-operator-aux-container) before running in the 4.0 operators. I don't see a need to keep the old functions in place and unchanged.

Reply:
"The 3.x auxiliary image pods will always be converted to "raw" init containers using special names"

The old "already running" pods could theoretically still be there, right? And it's not clear to me if a subsequent external pod-restart will cause them to run through the webhook, or simply restart the pod without operator/webhook intervention. So it seems risky to change the scripts out from underneath these pods...

Reply:
If the pod is externally deleted by the customer, it'll be recreated by the Operator based on converted domain resource yaml i.e. using the specs containing "raw" init containers with special names.

Reply:
How about if a pod restarts for other reasons? Like out-of-memory, etc.

Reply:
The pod restart due to out-of-memory should be same as customer deleting the pod externally. The pod will be recreated with new specs  with the "raw" init containers. If the container is restarted (for e.g. due to liveness probe failure), then the init container will not be run again.